webapp/packages/plugin-sql-editor/src/SqlEditor.tsx

```diff
@@ -37,15 +40,30 @@ export const SqlEditor = observer<Props>(function SqlEditor({ state }) {
 
   const opened = dataSource?.isOpened() || false;
 
+  const split = useSplit();
+  const data = useSqlEditor(state);
+  const sqlEditorModeService = useService(SqlEditorModeService);
+  const displayedEditors = getComputed(() => sqlEditorModeService.tabsContainer.getDisplayed({ state, data }).length);
+  const isEditorEmpty = displayedEditors === 0;
+
+  useEffect(() => {
+    if (isEditorEmpty) {
+      split.fixate('maximize', true);
+    } else if (split.state.disable) {
+      split.fixate('resize', false);
+      split.state.setSize(-1);
+    }
+  }, [isEditorEmpty]);
```

Comment:
I moved this logic in one place cause it looks weird when it is separate and harder to nail its behavior